packages/gatsby/src/internal-plugins/partytown/gatsby-ssr.tsx

```diff
@@ -0,0 +1,46 @@
+import React from "react"
+import type { GatsbySSR } from "gatsby"
+import { Partytown } from "@builder.io/partytown/react"
+import { PartytownContext } from "gatsby-script"
+import type { PartytownProps } from "@builder.io/partytown/react"
+
+const collectedScripts: Record<string, Array<PartytownProps>> = {}
+
+export const wrapRootElement: GatsbySSR[`wrapRootElement`] = ({
+  element,
+  pathname,
+}) => (
+  <PartytownContext.Provider
+    value={{
+      collectScript: (newScript: PartytownProps): void => {
+        collectedScripts[pathname] = [
+          ...(collectedScripts?.[pathname] || []),
+          newScript,
+        ]
```

Comment:
Let's not recreate arrays and let's just push to it (similar as https://github.com/gatsbyjs/gatsby/pull/35403/files#r870103603)